metals/src/main/scala/scala/meta/internal/parsing/DocumentSymbolProvider.scala

```diff
@@ -250,6 +250,73 @@ class DocumentSymbolProvider(trees: Trees) {
             t.name.pos,
             ""
           )
+        case t: Defn.Enum =>
+          addChild(
+            t.name.value,
+            SymbolKind.Enum,
+            t.pos,
+            t.name.pos,
+            ""
+          )
+          newOwner()
+        case t: Defn.RepeatedEnumCase =>
+          t.cases.foreach { name =>
+            addChild(
+              name.value,
+              SymbolKind.EnumMember,
+              name.pos,
+              name.pos,
+              ""
+            )
+          }
+          newOwner()
+        case t: Defn.EnumCase =>
+          addChild(
+            t.name.value,
+            SymbolKind.EnumMember,
+            t.pos,
+            t.name.pos,
+            ""
+          )
+        case t: Defn.ExtensionGroup =>
+          val (name, pos) =
+            t.eparam.decltpe.fold(("", t.pos))(tpe => (tpe.syntax, tpe.pos))
+          addChild(
+            s"extension $name",
+            SymbolKind.Class,
+            t.pos,
+            pos,
+            ""
+          )
+          newOwner()
+        case t: Defn.GivenAlias =>
+          addChild(
+            t.name.value,
+            SymbolKind.Constant,
+            t.pos,
+            t.name.pos,
+            t.decltpe.syntax
+          )
+          newOwner()
+        case t: Defn.Given =>
+          val (name, pos) =
+            t.name match {
+              case Name.Anonymous() =>
+                t.templ.inits match {
+                  case Nil => ("(anonymous)", t.pos)
+                  case inits =>
+                    (inits.map(_.tpe.syntax).mkString(" with "), inits.head.pos)
+                }
+              case name => (name.value, name.pos)
+            }
+          addChild(
+            name,
+            SymbolKind.Class,
+            t.pos,
+            pos,
+            ""
```

Comment:
Let's put the type here always as details if name is specified? Or always just keep it here and have name empty in case of anonymous givens?

Reply:
Added details. I don't think that there is a need to keep an empty name for anonymous instances.
Attached screenshot of how it looks now:
![Screenshot from 2021-02-05 20-11-57](https://user-images.githubusercontent.com/5816952/107066389-f974f080-67ee-11eb-90de-11d380579ece.png)

Reply:
It looks a bit weerd to have `Ord[String] Ord[String` - we should not duplicate information unless absolutely needed.

Reply:
That's fun because I've added it to show that it isn't weird at all =)
Ok, changed!